core/src/main/java/hudson/FilePath.java

```diff
@@ -507,6 +519,28 @@ public Integer invoke(File f, VirtualChannel channel) throws IOException {
 
             private static final long serialVersionUID = 1L;
     }
+    private class Archive2 extends SecureFileCallable<List<String>> {
```

Comment:
nit: Would it make sense to replace the implementation of `Archive` with the implementation in this new class and have the old `archive` method call `.size()` on the result to maintain the previous behavior, or would the result be slightly different in some cases? Not really important; it would just get rid of a bit of duplication.

Reply:
this sense backwards.
it would cause GC churn, and these would Iiuc have to be sent over to r remoting channel too iiuc